src/modules/accommodations/accommodations.service.ts

```diff
@@ -63,6 +73,99 @@ export class AccommodationsService {
     };
   }
 
+  async createAccommodation(
+    createDto: CreateAccommodationDto,
+    userId: string,
+  ): Promise<Accommodation> {
+    const { amenities, categoryId, ...accommodationData } = createDto;
+    const allAmenities = await this.amenitiesService.getAllAmenities();
+    const resolvedAmenities = amenities
+      ? allAmenities.filter((amenity) => amenities.includes(amenity.id))
+      : [];
+
+    const allCategories = await this.categoryService.getAllCategories();
+    const resolvedCategory = allCategories.find(
+      (category) => category.id === categoryId,
+    );
```

Comment:
You don't have to get all categories just to check if one exists. Search categoryId directly from database instead.